src/backend/storage/freespace/freespace.c

```diff
@@ -637,10 +637,13 @@ fsm_extend(Relation rel, BlockNumber fsm_nblocks)
 
 	while (fsm_nblocks_now < fsm_nblocks)
 	{
-		PageSetChecksumInplace((Page) pg.data, fsm_nblocks_now);
-
-		smgrextend(rel->rd_smgr, FSM_FORKNUM, fsm_nblocks_now,
-				   pg.data, false);
+		/* ZENITH: Initialize FSM pages through buffer cache to prevent loading them from pageserver */
```

Comment:
Why is this change needed? Is it still needed with the PD_WAL_LOGGED removal?

Reply:
Hm, @knizhnik, am I correct that with `smgrextend` this will hit `zenith_wallog_page` and freshly allocated FSM page will be WAL logged? And this doesn't have much sense, so this is a performance optimization, but is it related to the back pressure?